### DISCUSSION TITLE
Coder client workspace connectivity diagrams

### DISCUSSION THREAD

**Discussion by dcarrion87** [coder/coder#8792] [Announcements]
Are there any official diagrams we can point to that show how a coder client establishes a connection to a workspace and keeps the connection?

I think our diagrams are not giving the full picture as to what the client, workspace, agent are doing in establishing connectivity for a client and keeping connectivity.

I can only find  https://coder.com/docs/v2/latest/about/architecture which doesn't really give much of a detailed picture on the wireguard voodoo that is happening in the background.

**Comment by sharkymark** [comment]
https://coder.com/docs/v2/latest/networking

and our CTO's original issue breaks it down

https://github.com/coder/coder/issues/2779

**Comment by dcarrion87** [comment]
@sharkymark thanks but unfortunately not what we're looking for.

Would be great to see something more formally documented given it's a big component of the tool.

It's just so difficult to explain to people the usual connectivity workflow. And it just doesn't add up given some of the issues we face.

**Comment by kylecarbs** [comment]
Makes sense. We'll do something here!

**Comment by edaily** [comment]
Agree that this would be really helpful. The tailscale stuff is very cool but being able to confidently communicate how this works to other engineers / teams is pretty tough at the moment.

**Comment by matifali** [comment]
We have it documented here. 
https://coder.com/docs/architecture/architecture#components-2
